doc/manual/manual.tex

```diff
@@ -4730,42 +4730,133 @@ \subsubsection{Using tracer particles}
  
-In comparison to \url{cookbooks/composition-active.prm}, the \texttt{Composition} section now contains variables that specify the \texttt{Compositional field method} and \texttt{Mapped particle properties}.  The \texttt{Compositional field method} parameter may be set to either \texttt{field} or \texttt{particles}, where \texttt{field} specifies that compositional fields are tracked by advected fields (default method).  
-
-In this case, we specify that the compositional fields are in fact tracked by particles.  In order for the material model to access information stored on the particles, however, we also need to specify that the initial composition of each particle is mapped to a new field created at each time step for use in the material model.  The name of this field will still correspond to those specified in \texttt{set Names of fields}.  
-
-The initial composition is mapped to a new field with the general phrase \texttt{field{\_}name{:} initial field{\_}name}, where \texttt{field{\_}name} corresponds to \texttt{upper} or \texttt{lower}.  Notably, the order in which field names are defined in \texttt{set Names of fields} will determine the compositional field index number referenced in a material model. Therefore, it is recommended for consistency that mapped particle properties are listed in the same order as in \texttt{set Names of fields} . 
+What this does is the following: It says that there will be two
+compositional fields, called \texttt{lower} and \texttt{upper}
+(because we will use them to indicate material that comes from either
+the lower or upper part of the domain). Next, the
+\texttt{Compositional field methods} states that each of these fields
+will be computed by interpolation from the particles (if we had left
+this parameter at its default value, \texttt{field}, for each field,
+then it would have solved an advection PDE in each time step, as we
+have done in all previous examples).
+
+In this case, we specify that both of the compositional fields are in
+fact interpolated from particle properties in each time step. How this
+is done is described in the fourth line. To understand it, it is
+important to realize that particles and fields have matching names: We
+have named the fields \texttt{lower} and \texttt{upper}, whereas the
+properties that result from the \texttt{initial composition} entry in
+the particles section are called \texttt{initial lower} and
+\texttt{initial upper}, since they inherit the names of the fields.
+
+The syntax for interpolation from particles to fields then
+states that the \texttt{lower} field will be set to the interpolated
+value of the \texttt{initial lower} particle property, and similarly
+for the \texttt{lower} field. This seems circular, since the
+\texttt{initial composition} particle property was initialized from
+the fields. The relevant piece of information is that fields are
+\textit{always} initialized as fields before the first time step,
```

Comment:
This last part is not quite right. We use the same algorithm that initializes the compositional fields, but we do not project first onto the compositional field and evaluate then at the particle position, but rather call the compositional initial condition function with the particle position instead of the quadrature point position. This allows us to assign particle properties that are not representable by the compositional finite element (gradients are not diffused before being assigned to the particles). I think the real benefit of using the compositional initial condition plugins is that we do not need to duplicate these routines as particle initialization routines. It also allows us to interpret every advection mechanism as a 'compositional field'. Some of those fields are advected by solving and advection-diffusion PDE, some are advected by advecting particles, but both are initialized from the same initial condition.